diff --git a/build.gradle b/build.gradle
index 553079b6..751ac6a3 100644
--- a/build.gradle
+++ b/build.gradle
@@ -3,6 +3,7 @@ buildscript {
     repositories {
         jcenter()
         google()
+        maven { url System.getenv("GRADLE_MAVEN_REPO") }
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.1.4'
@@ -15,5 +16,6 @@ allprojects {
         maven { url "https://raw.githubusercontent.com/guardianproject/gpmaven/master" }
         google()
         maven { url 'https://jitpack.io' }
+        maven { url System.getenv("GRADLE_MAVEN_REPO") }
     }
 }
