From 820f26810baf35c1961763f6b0013e4f7e7380da Mon Sep 17 00:00:00 2001
From: Georg Koppen <gk@torproject.org>
Date: Thu, 12 Apr 2018 13:10:13 +0000
Subject: [PATCH] Use our gcc/c++


diff --git a/src/librustc_binaryen/build.rs b/src/librustc_binaryen/build.rs
index f23ff3cee5..2496dd1be9 100644
--- a/src/librustc_binaryen/build.rs
+++ b/src/librustc_binaryen/build.rs
@@ -26,6 +26,8 @@ fn main() {
 
     Config::new("../binaryen")
         .define("BUILD_STATIC_LIB", "ON")
+        .define("CMAKE_C_COMPILER", "gcc")
+        .define("CMAKE_CXX_COMPILER", "c++")
         .build_target("binaryen")
         .build();
 
-- 
2.16.3

