From 4402f247643c7b3b730f0facaf1c45dc02d5721f Mon Sep 17 00:00:00 2001
From: sisbell <shane.isbell@gmail.com>
Date: Thu, 16 May 2019 21:33:06 +0000
Subject: [PATCH] Fixes thaliproject#120: Tor Startup Broken on KitKat Devices


diff --git a/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfig.java b/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfig.java
index 3df0f12..1ab7c2d 100644
--- a/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfig.java
+++ b/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfig.java
@@ -372,7 +372,7 @@ public final class TorConfig {
         public TorConfig build() {
             if(homeDir == null) {
                 String userHome = System.getProperty("user.home");
-                homeDir = (userHome != null && !"".equals(userHome)) ? new File(userHome) : configDir;
+                homeDir = (userHome != null && !"".equals(userHome) && !"/".equals(userHome)) ? new File(userHome) : configDir;
             }
 
             if (torExecutableFile == null) {
-- 
2.20.1

