diff --git a/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfigBuilder.java b/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfigBuilder.java
index 731fb3e..e0d0c42 100644
--- a/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfigBuilder.java
+++ b/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfigBuilder.java
@@ -395,6 +395,9 @@ public final class TorConfigBuilder {
         if (!isNullOrEmpty(isolationFlag)) {
             buffer.append(" ").append(isolationFlag);
         }
+        buffer.append(" KeepAliveIsolateSOCKSAuth");
+        buffer.append(" IPv6Traffic");
+        buffer.append(" PreferIPv6");
         buffer.append('\n');
         return this;
     }
