diff --git a/android/build.gradle b/android/build.gradle
index b6d8608..38093be 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -7,6 +7,7 @@ buildscript {
         mavenLocal()
         mavenCentral()
         google()
+        maven { url System.getenv("GRADLE_MAVEN_REPO") }
     }
     dependencies {
         classpath "com.android.tools.build:gradle:${androidplugin}"
@@ -82,6 +83,7 @@ repositories {
     mavenCentral()
     google()
     maven { url "https://raw.githubusercontent.com/guardianproject/gpmaven/master" }
+    maven { url System.getenv("GRADLE_MAVEN_REPO") }
 }
 
 /**
diff --git a/android_tor_installer/build.gradle b/android_tor_installer/build.gradle
index 1bc0343..7d8e886 100644
--- a/android_tor_installer/build.gradle
+++ b/android_tor_installer/build.gradle
@@ -7,6 +7,7 @@ buildscript {
         mavenLocal()
         mavenCentral()
         google()
+        maven { url System.getenv("GRADLE_MAVEN_REPO") }
     }
     dependencies {
         classpath "com.android.tools.build:gradle:${androidplugin}"
@@ -50,5 +51,6 @@ repositories {
     google()
     maven { url "https://repo.spring.io/plugins-release" }
     maven { url "https://raw.githubusercontent.com/guardianproject/gpmaven/master" }
+    maven { url System.getenv("GRADLE_MAVEN_REPO") }
 
 }
diff --git a/build.gradle b/build.gradle
index a269024..430f3e1 100644
--- a/build.gradle
+++ b/build.gradle
@@ -9,6 +9,7 @@ subprojects {
     mavenLocal()
     mavenCentral()
     maven { url "https://repo.spring.io/plugins-release" }
+    maven { url System.getenv("GRADLE_MAVEN_REPO") }
   }
 }
 
