diff -ur nsis-2.51-src/SCons/Config/gnu nsis-2.51-src.n/SCons/Config/gnu
--- nsis-2.51-src/SCons/Config/gnu	2018-05-24 20:40:06.508000000 +0200
+++ nsis-2.51-src.n/SCons/Config/gnu	2018-05-24 21:04:16.501000000 +0200
@@ -89,6 +89,7 @@
 stub_env.Append(LINKFLAGS = ['$ALIGN_FLAG'])        # 512 bytes align
 stub_env.Append(LINKFLAGS = ['-Wl,-e,_WinMain@16']) # entry point
 stub_env.Append(LINKFLAGS = ['$MAP_FLAG'])          # generate map file
+stub_env.Append(LINKFLAGS = ['-Wl,--no-insert-timestamp']) # remove timestamps for reproducible builds
 
 ### makensis environment
 
@@ -125,6 +126,7 @@
 plugin_env.Append(LINKFLAGS = ['$MAP_FLAG'])        # generate map file
 plugin_env.Append(LINKFLAGS = ['-static-libgcc'])   # Make sure libgcc is statically linked in, for the plugins to work.
 plugin_env.Append(LINKFLAGS = ['-static-libstdc++']) # Make sure libstdc++ is statically linked in, for the plugins to work.
+plugin_env.Append(LINKFLAGS = ['-Wl,--no-insert-timestamp']) # remove timestamps for reproducible builds
 
 ### cross-platform util environment
 
@@ -148,6 +150,7 @@
 
 util_env.Append(LINKFLAGS = ['-mwindows'])          # build windows executables
 util_env.Append(LINKFLAGS = ['$ALIGN_FLAG'])        # 512 bytes align
+util_env.Append(LINKFLAGS = ['-Wl,--no-insert-timestamp']) # remove timestamps for reproducible builds
 
 conf = FlagsConfigure(util_env)
 if not defenv['DEBUG'] and defenv['STRIP'] and defenv['STRIP_W32']:
