From 2d7897d479a13cbf68335140b61228cd5552dfd9 Mon Sep 17 00:00:00 2001
From: David Fifield <david@bamsoftware.com>
Date: Mon, 3 Dec 2018 21:14:32 -0700
Subject: [PATCH] Disable some settings in build_overrides.

---
 build_overrides/build.gni | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build_overrides/build.gni b/build_overrides/build.gni
index 81cb3e73ae..5304a64ce8 100644
--- a/build_overrides/build.gni
+++ b/build_overrides/build.gni
@@ -10,7 +10,7 @@
 enable_java_templates = true
 
 # Some non-Chromium builds don't use Chromium's third_party/binutils.
-linux_use_bundled_binutils_override = true
+linux_use_bundled_binutils_override = false
 
 # Variable that can be used to support multiple build scenarios, like having
 # Chromium specific targets in a client project's GN file etc.
-- 
2.11.0

