diff --git a/mobile/android/app/build.gradle b/mobile/android/app/build.gradle
index c8380042d8a9..990fabe05a60 100644
--- a/mobile/android/app/build.gradle
+++ b/mobile/android/app/build.gradle
@@ -225,8 +225,10 @@ dependencies {
     implementation "com.android.support:design:${mozconfig.substs.ANDROID_SUPPORT_LIBRARY_VERSION}"
     implementation "com.android.support:customtabs:${mozconfig.substs.ANDROID_SUPPORT_LIBRARY_VERSION}"
     implementation "com.android.support:palette-v7:${mozconfig.substs.ANDROID_SUPPORT_LIBRARY_VERSION}"
-    implementation files('orbotservice-release.aar')
+    implementation files('android-release.aar')
+    implementation files('universal-0.0.3.jar')
     implementation files('jsocksAndroid-release.aar')
+    implementation files('service-release.aar')
 
     if (mozconfig.substs.MOZ_NATIVE_DEVICES) {
         implementation "com.android.support:mediarouter-v7:${mozconfig.substs.ANDROID_SUPPORT_LIBRARY_VERSION}"
@@ -265,9 +267,10 @@ dependencies {
     // Including the Robotium JAR directly can cause issues with dexing.
     androidTestImplementation 'com.jayway.android.robotium:robotium-solo:5.5.4'
 
-    // Orbotservice
-    implementation 'org.torproject:tor-android-binary:0.3.4.9'
-    implementation 'com.jrummyapps:android-shell:1.0.1'
+    //tor-android-service
+    implementation 'net.freehaven.tor.control:jtorctl:0.2'
+    implementation 'org.slf4j:slf4j-api:1.7.25'
+    implementation 'org.slf4j:slf4j-android:1.7.25'
 }
 
 // TODO: (bug 1261486): This impl is not robust -
